Frameworks/CoreFoundation/PlugIn.subproj/CFBundle_Locale.c

```diff
@@ -663,8 +673,66 @@ static CFStringRef _CFBundleCopyLanguageFoundInLocalizations(CFArrayRef localiza
+#if DEPLOYMENT_TARGET_WINDOWS
+    // Here we need to intersect the preferred languages with the available localizations
+    // We know the user languages are in preferred order, add to this list in this order
+    // Prefer the full language locale, attempt to convert any hyphens to underscores as
+    // Windows language settings are retrieved with hyphens while underscores are commonly used for localization.
+    // Finally, attempt to truncate any underscores from the language to find a base localization.
+
+    CFMutableArrayRef resultArray = CFArrayCreateMutable(kCFAllocatorSystemDefault, 0, &kCFTypeArrayCallBacks);
+    
+    for (CFIndex i = 0, count = CFArrayGetCount(preferredLocalizations); i < count; i++) {
+        CFStringRef preferredLocalization = (CFStringRef)CFArrayGetValueAtIndex(preferredLocalizations, i);
+        for (CFIndex j = 0, count = CFArrayGetCount(availableLocalizations); j < count; j++) {
+            CFStringRef availableLocalization = (CFStringRef)CFArrayGetValueAtIndex(availableLocalizations, j);
+            if(CFStringCompare(preferredLocalization, availableLocalization, 0) == kCFCompareEqualTo) {
+                CFArrayAppendValue(resultArray, preferredLocalization);
+            }
+            CFRange hyphenation;
+            if (CFStringFindWithOptions(preferredLocalization, CFSTR("-"), CFRangeMake(0, CFStringGetLength(preferredLocalization)), kCFCompareCaseInsensitive, &hyphenation) == true) {
+                CFStringRef underscoreNotationLocalization = _copyStringWithUnderscores(preferredLocalization);
+                if (CFStringCompare(underscoreNotationLocalization, availableLocalization, 0) == kCFCompareEqualTo) {
+                    CFArrayAppendValue(resultArray, underscoreNotationLocalization);
+                }
+
+                CFStringRef truncatedLocalization = _copyStringTruncated(underscoreNotationLocalization, hyphenation);
+                if (CFStringCompare(truncatedLocalization, availableLocalization, 0) == kCFCompareEqualTo) {
+                    CFArrayAppendValue(resultArray, truncatedLocalization);
+                }
+
+                CFRelease(underscoreNotationLocalization);
+                CFRelease(truncatedLocalization);
+            } else if (CFStringFindWithOptions(preferredLocalization, CFSTR("_"), CFRangeMake(0, CFStringGetLength(preferredLocalization)), kCFCompareCaseInsensitive, &hyphenation) == true) {
+                CFStringRef truncatedLocalization = _copyStringTruncated(preferredLocalization, hyphenation);
+                if (CFStringCompare(truncatedLocalization, availableLocalization, 0) == kCFCompareEqualTo) {
+                    CFArrayAppendValue(resultArray, truncatedLocalization);
+                }
+                CFRelease(truncatedLocalization);
+            }
+        }
+    }
+    if (CFArrayGetCount(resultArray) > 0) {
+        return resultArray;
+    }
+    return NULL;
```

Comment:
it looks like the codepath below has fallbacks for `en_US` and `devlang` (they've called them "backstops").

Reply:
Yup. I'm not sure what you mean by this though.

Reply:
what I mean by this is that you're behoven to do the same ;)

Reply:
Mmm, no? We want to return NULL so that the rest of the code will hit it's fallbacks. Where it creates "devLangArray" and an "en_US" array to call _CFBundleCreateMutableArrayOfFallbackLanguages.

Sorry if I'm missing something.

Reply:
Thanks to me viewing code in github's web UI, I didn't see that the backstop is in a completely different function (the one that calls this one), and that returning is the correct thing to do.